pkg/controller/utils/elasticsearch.go

```diff
@@ -210,7 +210,10 @@ func (es *esClient) createOrUpdatePolicies(ctx context.Context, listPolicy map[s
 
 		res, err := es.client.XPackIlmGetLifecycle().Policy(policyName).Do(ctx)
 		if err != nil {
-			if eerr, ok := err.(*elastic.Error); ok && eerr.Status == 404 {
+			if elastic.IsConnErr(err) {
+				// If connection error, reset elasticsearch client
+				es.client = nil
```

Comment:
Where this is created there is a Lock that is held when the client is created? It makes me think when setting this to nil the same Lock should be obtained.